client/client.go

```diff
@@ -162,6 +162,11 @@ type Client interface {
 	// this may differ from the initial Endpoints provided in the Config.
 	Endpoints() []string
 
+	// RestEndpoints resets the set of API endpoints used by Client to resolve
```

Comment:
I dunno, how is it more specific? To me `Reset` has more of an implication that it's reverting to some default/initial state, whereas `x.Set(y)` is very explicitly "set the state of x to y"

I don't feel extremely strongly about this but curious about your reasoning.